src/pudl/transform/classes.py

```diff
@@ -883,18 +885,30 @@ def spot_fix_values(self, df: pd.DataFrame, params: SpotFixes) -> pd.DataFrame:
     Returns:
         The same input DataFrame but with some spot fixes corrected.
     """
-    # The default SpotFixValues() instance has no effect:
-    if params.record_id is None or params.record_col is None:
-        return df
+    spot_fixes_df = pd.DataFrame(
+        params.spot_fixes, columns=params.idx_cols + params.fix_cols
+    )
 
-    elif params.record_col in list(df.columns):  # If record_col in df
-        for key in params.fixes:  # For each fix
-            if key in list(df.columns):
-                df.loc[df[params.record_col] == params.record_id, key] = params.fixes[
-                    key
-                ]  # Manually update value
+    if not (spot_fixes_df.dtypes == df[params.idx_cols + params.fix_cols].dtypes).all():
```

Comment:
I think this may be an overly strict test. E.g. what if the type of the specified fix is `int` but the column being fixed a nullable pandas `Int64` dtype?  Some unit tests will help ferret out this kind of thing too.

Reply:
Yeah that sounds like a good solution -- just making sure the fix is compatible (convertible to) the type that it'll find in the context of the dataframe it's part of.